README.md

```diff
@@ -1,2 +1,494 @@
+| Offers   | Offers for Mumbai location     | This test mocks the GPS location for Mumbai and verifies that the product offers applicable for the Mumbai location are shown.   |
+| Product  | Apply Apple Vendor Filter          | This test verifies that 9 Apple products are only shown if the Apple vendor filter option is applied. |
+| Product  | Apply Lowest to Highest Order By   | This test verifies that the product prices are in ascending order when the product sort "Lowest to Highest" is applied. |
+| User     | Login as User with no image loaded | This test verifies that the product images load for user: "image_not_loading_user" on the e-commerce application. Since the images do not load, the test case assertion fails.|
+| User     | Login as User with existing Orders |  This test verifies that existing orders are shown for user: "existing_orders_user"  |
+  
+---
+
+
+## Test infrastructure environments
+
+- [On-premise/self-hosted](#on-premise-self-hosted)
+- [Docker](#docker)
+- [BrowserStack](#browserstack)
+
+
+## Configuring the maximum parallel test threads for this repository
+
+For all the parallel run configuration profiles, you can configure the maximum parallel test threads by changing the settings below.
+
+- Docker
+
+  [docker-compose.yml](docker-compose.yml)
+  
+  NODE_MAX_INSTANCES = 5
+  GRID_MAX_SESSION = 5
+ 
+
+- BrowserStack
+
+  - Maven:
+
+    [pom.xml](pom.xml)
+
+    parallel-count = 5
+
+
+## Test Reporting
+
+- [Allure reports](#generating-allure-reports)
+
+---
+
+# On Premise / Self Hosted
+
+This infrastructure points to running the tests on your own machine using a browser (e.g. Chrome) using the browser's driver executables (e.g. ChromeDriver for Chrome). Selenium enables this functionality using WebDriver for many popular browsers.
+
+## Prerequisites
+
+- For this infrastructure configuration (i.e on-premise) ensure that the ChromeDriver executable is placed in the `/src/test/resources/` folder.
+
+Note: The ChromeDriver version must match the Chrome browser version on your machine.
+
+## Running Your Tests
+
+### Run a specific test on your own machine
+
+- How to run the test?
+
+  To run the default test scenario (e.g. End to End Scenario) on your own machine, use the following command:
+
+  Maven:
+    ```sh
+  mvn test -P on-prem
+  ```
+
+  Gradle:
+    ```sh 
+  gradle on-prem
+  ```
+
+  To run a specific test scenario, use the following command with the additional 'test-name' argument:
+
+  Maven:
+  ```sh
+  mvn test -P on-prem -Dtest-name="<Test scenario name>"
+  ```
+
+  Gradle:
+  ```sh
+  gradle on-prem -Dtest-name="<Test scenario name>"
+  ```
+
+  where,  the argument 'test-name' can be any Cucumber scenario name configured in this repository.
+
+  E.g. "Login as username", "Login as Locked User", "Offers for mumbai geo-location" or any of the other test scenario names, as outlined in [About the tests in this repository](#About-the-tests-in-this-repository) section.
```

Comment:
`mvn test -P on-prem -Dtest-name="Offers for mumbai geo-location"`
`mvn test -P on-prem -Dtest-name="Login as username"`

Both these profiles do not trigger any test.

Reply:
The profile names are different than the one's mentioned in the README.md
This is the right command:
 `mvn test -P on-prem -Dtest-name="Login with given '<username>' user" `
` mvn test -P on-prem -Dtest-name="Offers for Mumbai location"`        
We will make the changes.